twap/price/price.go

```diff
@@ -0,0 +1,48 @@
+package price
+
+import (
+	"fmt"
+	"sort"
+	"time"
+)
+
+type Price struct {
+	Value     float64   `json:"price"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
+func TWAP(samples []Price) (float64, error) {
+	if len(samples) == 0 {
+		return 0, fmt.Errorf("no samples provided")
+	}
+	if len(samples) == 1 {
+		return samples[0].Value, nil
+	}
+
+	// Sort samples from latest to earliest
+	sort.Slice(
```

Comment:
FWIW, to make it clear what is going on, I would suggest updating this to:
```
lessThan := func(i, j int) bool { return samples[i].Timestamp.After(samples[j].Timestamp) }
sort.Slice(samples, lessThan)
```

Reply:
will update